### ISSUE TITLE
Wrong `fs.errors.CreateFailed: Root path "XXX" does not exist`

### ISSUE THREAD

**Issue by yonromai** [Othoz/gcsfs#11]
Hi,

When doing:
```python
fs = open_fs("gs://my_bucket/path/to/blob")
```
I'm getting:
```
  File "/root/.pyenv/versions/3.6.8/lib/python3.6/site-packages/fs/opener/registry.py", line 228, in open_fs
    default_protocol=default_protocol,
  File "/root/.pyenv/versions/3.6.8/lib/python3.6/site-packages/fs/opener/registry.py", line 189, in open
    open_fs = opener.open_fs(fs_url, parse_result, writeable, create, cwd)
  File "/root/.pyenv/versions/3.6.8/lib/python3.6/site-packages/fs_gcsfs/opener.py", line 25, in open_fs
    return GCSFS(bucket_name, root_path=root_path, create=create)
  File "/root/.pyenv/versions/3.6.8/lib/python3.6/site-packages/fs_gcsfs/_gcsfs.py", line 88, in __init__
    raise errors.CreateFailed("Root path \"{}\" does not exist".format(root_path))
fs.errors.CreateFailed: Root path "path/to/blob" does not exist
```
But I'm sure `gs://my_bucket/path/to/blob` does exist (running `gsutil ls gs://my_bucket/path/to/blob` returns files)

Wondering if https://github.com/Othoz/gcsfs/pull/9 broke something cc @birnbaum

**Comment by birnbaum** [comment]
Hi! As `strict=True` by default, GCSFSexpects a directory marker (an empty file with the name `path/to/blob/`) at the root path, see https://fs-gcsfs.readthedocs.io/en/latest/#limitations. I guess this file is missing in your case as the data in your bucket was not created via GCSFS. 

You can fix this via the [fix_storage()](https://fs-gcsfs.readthedocs.io/en/latest/#fs_gcsfs.GCSFS.fix_storage) utility method. Let me know if it worked :)

**Comment by yonromai** [comment]
@birnbaum Thanks for coming back to me quickly. 


I see, good to know - sorry for the noise. I would want to avoid adding files in there since the content of these folders is read using globing (by apache beam).

Indeed, I got it working by manually disabling the strict mode `GCSFS(..., ..., strict=False)`.
Btw: is `gs://mybucket/root_path?strict=False` supposed to disable the strict check? doesn't seem to be working

**Comment by birnbaum** [comment]
I know, the additional files are annoying but unfortunately there is no other way. You could filter out all blobs ending with "/" as this is the convention for "directories" in GCS and S3 (they are displayed as such in the web frontends, too). But I see it's not very clean.

Yes, I would expect `gs://mybucket/root_path?strict=False` to work, but I never tried it out. I will not have time to look into this before monday though, sorry! I will get back to you then

**Comment by birnbaum** [comment]
@yonromai The `gs://mybucket/root_path?strict=False` issue is fixed in 0.4.2, it's now published!
